src/ir/ir.jl

```diff
@@ -158,10 +244,48 @@ arguments(ir::IR) = arguments(block(ir, 1))
 argtypes(b::Block) = argtypes(BasicBlock(b))
 argtypes(ir::IR) = argtypes(block(ir, 1))
 
+
+"""
+    canbranch(b::Block)
+
+Check whether the block `b` will branch either implicitly (by fallthrough) or explicitly 
+(by the last branch being conditional).
```

Comment:
I would describe this as "adding a new branch would be valid" (since if there's a non-conditional branch, adding a new one would be redundant).

Reply:
Ok, I just understood this differently.  But then, the name is a bit deceptive to me -- although I can't come up with a better one.

Reply:
I think it's not exported at the moment, and should probably stay that way. If someone wants this it's easiest for them to write it down themselves.

Reply:
OK, should I leave it as it is then? Or remove the entry from `reference.md`?

Reply:
Hmm it is actually exported. We can deprecate it separately if we want to.